rules/apps.libsonnet

```diff
@@ -84,23 +84,18 @@
         name: 'k8s.rules.container_memory_requests',
         rules: [
           {
-            record: 'cluster:namespace:pod_memory:active:kube_pod_container_resource_requests',
+            record: 'cluster:namespace:pod_memory:active:kube_pod_resource_request_or_kube_pod_container_resource_requests',
             expr: |||
-              kube_pod_container_resource_requests{resource="memory",%(kubeStateMetricsSelector)s}  * on (namespace, pod, %(clusterLabel)s)
-              group_left() max by (namespace, pod, %(clusterLabel)s) (
-                (kube_pod_status_phase{phase=~"Pending|Running"} == 1)
-              )
+              (kube_pod_resource_request{resource="memory",%(kubeSchedulerSelector)s} or kube_pod_container_resource_requests{resource="memory",%(kubeStateMetricsSelector)s})
             ||| % $._config,
           },
           {
-            record: 'namespace_memory:kube_pod_container_resource_requests:sum',
+            record: 'namespace_memory:kube_pod_resource_request_or_kube_pod_container_resource_requests:sum',
             expr: |||
               sum by (namespace, %(clusterLabel)s) (
                   sum by (namespace, pod, %(clusterLabel)s) (
                       max by (namespace, pod, container, %(clusterLabel)s) (
-                        kube_pod_container_resource_requests{resource="memory",%(kubeStateMetricsSelector)s}
-                      ) * on(namespace, pod, %(clusterLabel)s) group_left() max by (namespace, pod, %(clusterLabel)s) (
-                        kube_pod_status_phase{phase=~"Pending|Running"} == 1
+                        kube_pod_resource_request{resource="memory",%(kubeSchedulerSelector)s} or kube_pod_container_resource_requests{resource="memory",%(kubeStateMetricsSelector)s}
```

Comment:
I'm not quite sure that it will do the right thing since `kube_pod_resource_request` and `kube_pod_container_resource_requests` don't have the exact same labels if I understand correctly.

Reply:
I could be wrong here but wouldn't the differing labelsets (kube-scheduler metrics having the additional `scheduler` and `priority` labels) after `or` be sanitized to the set of specified labels in the `max` operation, so doing an `ignoring (scheduler,priority)` will eventually have no effect on the final result?